pkg/topolvm-scheduler/cmd/root.go

```diff
@@ -80,23 +87,31 @@ func subMain() error {
 		return err
 	}
 
-	serv := &well.HTTPServer{
-		Server: &http.Server{
-			Addr:    config.ListenAddr,
-			Handler: h,
-		},
+	serv := &http.Server{
+		Addr:        config.ListenAddr,
+		Handler:     accessLogHandler(parentCtx, h),
+		ReadTimeout: 30 * time.Second,
 	}
 
-	err = serv.ListenAndServe()
-	if err != nil {
-		return err
-	}
-	err = well.Wait()
+	var wg sync.WaitGroup
+	defer wg.Wait()
```

Comment:
Maybe it's a matter of preference, but you don't have to wait for goroutine to terminate using wg.